src/Login.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react'
+import {
+  Grid,
+  Row,
+  Col,
+  FormGroup,
+  ControlLabel,
+  FormControl,
+  HelpBlock,
+  Button,
+  Alert,
+  Form
+} from 'react-bootstrap'
+import { withRouter } from 'react-router'
+
+import NavBar from './components/NavBar'
+import { Api } from './utils'
+import { FieldGroup } from './components'
+
+class Login extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      error: null,
+      username: '',
+      password: ''
+    }
+    this.onChangeUsername = this.onChangeUsername.bind(this)
+    this.onChangePassword = this.onChangePassword.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+  }
+
+  onChangeUsername(inputUsername) {
+    this.setState({
+      username: inputUsername.target.value
+    })
+  }
+
+  onChangePassword(inputPassword) {
+    this.setState({
+      password: inputPassword.target.value
+    })
+  }
+
+  onSubmit() {
+    const { username, password } = this.state
+    const data = {
+      username: username,
+      password: password
+    }
+    Api.postLogin(data)
+      .then(({ message: error, status }) => {
+        if (status === 'failed') {
+          this.setState({ error })
+        } else {
+          this.props.history.push('/', { isEditing: true })
```

Comment:
Instead of a successful login turning on edit mode, have it set a state for being logged in. And then show the ability to go into edit mode if the user is logged in